### DISCUSSION TITLE
URL Hash to select specific speakers and or play lists

### DISCUSSION THREAD

**Discussion by rxritalin** [music-assistant/support#2346] [Feature requests and ideas]
Would like to request URL Hashes for the web view in music assistant. 

The idea I am trying to accomplish, though there are other uses, is to have an NFC on or near a set of speakers. 
Guests that do not have access to my home-assistant can then tap their phone on the NFC and be presented with a URL that directs them to my Music Assistant web-view with the correct speaker already selected. 
Bonus, if the URL hash can also contain a playlist. 

This would also simplify using the web view in Lovelace as we would be able to have the correct speakers for the rooms UI already selected and ready to play.

**Comment by rxritalin** [comment]
Ok, so it seems playlist hashes are already a thing. But cant figure out a way to do it for selecting a speaker.

**Comment by TinyShark** [comment]
+! for this

**Comment by OzGav** [comment]
This has been implemented in 2.1.0 stable and later. Docs to follow but here are the instructions for now https://github.com/music-assistant/frontend/pull/585

**Reply by samwise22** [reply]
This is so nearly what I was hoping for but I don't want the full screen player window. I want the target page (home/search/playlists etc) to open but with the player specified in the URL pre-selected. Is there a way to do that?

**Reply by OzGav** [reply]
Yes for the view selection but you will have to expose port 8095. Review the docs for the pros and cons for that. Then just navigate to one of the veiws and take note of the address in your browser address bar, As for the player pre-selected that is coming in the next server update.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
